wcfsetup/install/files/lib/system/Environment.class.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace wcf\system;
+
+/**
+ * Provides functions to interact with the system/hosting enviroment.
+ *
+ * @author  Tim Duesterhus
+ * @copyright   2001-2021 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\System
+ */
+final class Environment
+{
+    public const SYSTEM_ID_REGISTRY_KEY = self::class . "::getSystemId()";
+
+    /**
+     * Returns a string representing key system information where
+     * changes might affect the software's behavior.
+     *
+     * Current information includes the location within the file system,
+     * the PHP minor version and the MySQL fork and minor version.
+     */
+    public static function getSystemId(): string
+    {
+        $ctx = \hash_init('sha256');
+
+        $sqlVersion = WCF::getDB()->getVersion();
+
+        foreach (
+            [
+                // 1) The path to this class as a proxy for the document root.
+                \sprintf("__FILE__ %s", __FILE__),
+                // 2) The PHP minor version.
+                \sprintf('PHP %s', self::getMinorVersion(\PHP_VERSION)),
```

Comment:
```suggestion
                \sprintf('PHP %s.%s', \PHP_MAJOR_VERSION, \PHP_MINOR_VERSION),
```
See https://www.php.net/manual/en/reserved.constants.php#reserved.constants.core

Reply:
I initially considered that, but this would be inconsistent with `SystemCheck` itself:

https://github.com/WoltLab/WCF/blob/master/wcfsetup/install/files/lib/acp/page/SystemCheckPage.class.php#L351

Reply:
I see, then disregard my comment please. I'm not sure why we went the extra mile in the system check, but you are right about being consistent.